Lib/multiprocessing/resource_tracker.py

```diff
@@ -26,8 +32,14 @@
 _HAVE_SIGMASK = hasattr(signal, 'pthread_sigmask')
 _IGNORED_SIGNALS = (signal.SIGINT, signal.SIGTERM)
 
+_CLEANUP_FUNCS = {
+    'folder': shutil.rmtree,
```

Comment:
Is this one only used in tests?

Reply:
For now yes. This refers to the "public API" discussion that I evoked here https://github.com/python/cpython/pull/13222#issue-277494385. By default, folders (created using `os.mkdir` for example) are not tracked by the resource tracker. But some users may want to track the lifecycle of folders created during python sessions. If `resource_tracker.register` is made public, we can imagine users writing code such as:

```python
>>> import os
>>> from multiprocessing import resource_tracker
>>> fname = '/tmp/foo'
>>> os.mkdir(fname)
>>> resource_tracker.register(fname, "folder")
```

But the public API discussion should maybe not be done in this PR. This first PR is mostly for not leaking `shared_memory` objects, so I am fine with removing this line for the sake of issue separation.

Reply:
Yes, I think it's better to remove it. For the purposes of the PROBE command, you may add a "noop" resource type.